src/libpam.rs

```diff
@@ -382,6 +385,14 @@ impl PamLibExt for Pam {
     fn retrieve_bytes(&self, module_name: &str) -> PamResult<Vec<u8>> {
         unsafe { self.retrieve_data::<PamByteData>(module_name) }.map(|data_cb| data_cb.data)
     }
+
+    fn syslog(&self, lvl: LogLvl, msg: &str) -> PamResult<()> {
+        let cmsg = CString::new(msg)?;
+        unsafe {
+            pam_syslog(self.0, lvl as c_int, cmsg.as_ptr());
```

Comment:
```suggestion
            pam_syslog(self.0, lvl as c_int, b"%s\0".as_ptr() as *const i8, cmsg.as_ptr());
```
Because otherwise, if a user happens to pass a formatting character like `%s` or `%p` in the message we'll get the content of the stack printed to syslog, and potentially (hopefully) a segfault. If the message being logged is controlled by an untrusted source this could lead to bad things™

Reply:
Oof, makes perfect sense. Thanks for catching this.